kernel/acpi/src/lib.rs

```diff
@@ -79,6 +79,14 @@ pub fn init(page_table: &mut PageTable) -> Result<(), &'static str> {
         let acpi_tables = ACPI_TABLES.lock();
         if let Some(hpet_table) = hpet::HpetAcpiTable::get(&acpi_tables) {
             hpet_table.init_hpet(page_table)?;
+            let period = time::Period::new(
+                hpet::get_hpet()
+                    // Initialisation succedeed so this cannot fail.
+                    .expect("couldn't get HPET")
```

Comment:
the above call to `init_hpet()` returns a wrapped `Hpet` object, so you can just use that directly instead of making a fallible call to `get_hpet()` again.